pkg/kubelet/dockertools/docker.go

```diff
@@ -113,6 +115,22 @@ func parseImageName(image string) (string, string) {
 	return parsers.ParseRepositoryTag(image)
 }
 
+func filterHTTPError(err error, image string) error {
+	// docker/docker/pull/11314 prints detailed error info for docker pull.
+	// When it hits 502, it returns a verbose html output including an inline svg,
+	// which makes the output of kubectl get pods much harder to parse.
+	// Here converts such verbose output to a concise one.
+	jerr, ok := err.(*jsonmessage.JSONError)
+	if ok && (jerr.Code == http.StatusBadGateway ||
+		jerr.Code == http.StatusServiceUnavailable ||
+		jerr.Code == http.StatusGatewayTimeout) {
+		glog.V(2).Infof("Pulling image %s failed: %v", image, err)
+		return fmt.Errorf("image pull failed for %s because the registry is temporarily unavailbe.", image)
```

Comment:
nit: prefer %q to %s. It makes it clear what is a variable and helps when the string has whitespace